File: 프로그래머스/Programmers/Lv.2/기능개발.js
function solution(progresses, speeds) {
  var answer = [];
  let countArr = [];
  let num = 0;
  let count = 1;

  for (let i = 0; i < progresses.length; i++) {
    num = Math.ceil((100 - progresses[i]) / speeds[i]);
    countArr.push(num);
  }

  for (let i = 0; i < countArr.length; i++) {
    for (let j = i; j < countArr.length; j++) {
      if (countArr[i] >= countArr[j + 1]) {
        count++;
      } else {
        answer.push(count);
        count = 1;
        i = j;
        break;
      }
    }
  }

  return answer;
}

// 모든 progress의 값이 100이 될 때까지 수행을 하고 => 그냥 계산해도 될 것 같음
// progress의 값이 될 때까지 걸린 시간을 배열로 기록한다.
// 가장 앞에 있는 인덱스의 값보다 큰 값이 나타날 때까지 count한다
// + 자기보다 큰 값이 나타나면 count 값을 return에 담는다.

// Ceil 올림함수 사용 => 나누기 할 때는 생각을 좀 해야 함

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    index = 0
    
    while index < len(progresses):
        for i in range(0,len(progresses)):
            progresses[i] += speeds[i]
        
        nowCnt = 0
        
        while progresses[index] >= 100:
            nowCnt += 1
            index += 1
            if index == len(progresses):
                break
        if nowCnt != 0:
            answer.append(nowCnt)
    
    
    return answer

# 순서는 중요. 바꿀 수 없음
# 일단 맨 앞에 있는거 기준으로 progress를 계속 높이다가 100보다 크거나 같으면
# return에 카운팅만큼 추가